packages/content/src/types.ts

```diff
@@ -0,0 +1,137 @@
+import { IItem, IGroup } from "@esri/arcgis-rest-types";
+
+export type HubType = 'document' | 'dataset' | 'map' | 'app' | 'site' | 'initiative' | 'event' | 'template';
+
+// Based on the [Hub Content UI Inventory](https://app.lucidchart.com/invitations/accept/dab82a25-2ed9-4f63-8075-3db8873087e3)
+// Attempting a simple _Hungarian Notation_ for attributes suffix: `*Date`, `*Id`, `*Number`. Plural (e.g. `tags`) is an array.
+
+export interface IContent extends IItem {
+  
+  // Inherited metadata (noting relevant subset of all item attributes)
+  // id: string; // item.id
+  // title: string; // item.title
+  // snippet: string; // item.snippet
+  // description: string; // item.description
+  // categories?: Array<string>; // item.categories
+  // tags: Array<string>; // item.tags  
+  // culture: string; // written locale e.g. en, es, cn
+  // url?: string;
+  // size: number;
+
+  // Derived metadata
+  hubType: HubType;
+  maintainer: IHubMember; // item.owner with more user metadata
+  access: { // overrides item.access with more attributes. could flatten.
+    visibility: 'private' | 'org' | 'public'; // item.access
+    permission?: 'view' | 'edit' | 'admin'; // item.itemControl
```

Comment:
Is there a case where we could obtain a `IContent`  result and not have at least `view` permission?

Reply:
good point.

I think a bigger question is how will `permission` be set? Do we expect the Hub API to return this? If not, then I think we should remove it and have this be something that's part of the view model at the application level.

Reply:
as noted in the comments, the value of `permission` comes _directly_ from the AGO Portal API.  [see Item Doc and search for 'itemControl'](https://developers.arcgis.com/rest/users-groups-and-items/item.htm). 

The common/default value will be `view`. The only reason view would not be available is if the item is blocked. e.g. `getContent('Art Vandelay')` . but that's not relevant as the response to that should be a better error IMO.